src/python/pants/goal/run_tracker.py

```diff
@@ -38,65 +38,90 @@ def default(self, o):
         return super().default(o)
 
 
-class RunTracker(Subsystem):
+class DeprecatedRunTracker(Subsystem):
     options_scope = "run-tracker"
     help = "Tracks and times the execution of a pants run."
 
-    # The name of the tracking root for the main thread (and the foreground worker threads).
-    DEFAULT_ROOT_NAME = "main"
-
     @classmethod
     def register_options(cls, register):
         register(
             "--stats-local-json-file",
             advanced=True,
             default=None,
+            removal_version="2.3.0.dev0",
+            removal_hint=(
+                "This option is now a noop: instead, use global option " "`--stats-json-file`."
+            ),
             help="Write stats to this local json file on run completion.",
         )
         register(
             "--stats-option-scopes-to-record",
             advanced=True,
             type=list,
             default=["*"],
+            removal_version="2.3.0.dev0",
+            removal_hint=(
+                "This option is now a noop: instead, use global option "
+                "`--stats-record-option-scopes`."
+            ),
             help="Option scopes to record in stats on run completion. "
             "Options may be selected by joining the scope and the option with a ^ character, "
             "i.e. to get option `pantsd` in the GLOBAL scope, you'd pass `GLOBAL^pantsd`. "
             "Add a '*' to the list to capture all known scopes.",
         )
 
-    def __init__(self, *args, **kwargs):
+
+class RunTracker:
```

Comment:
Removing the `Subsystem` inheritance allowed mypy to check this more effectively for some reason... the vast majority of initialization was able to move to `__init__`, which avoided the need for a bunch of type annotations.